module-utils/log/LoggerBufferContainer.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#pragma once
+
+#include "LoggerBuffer.hpp"
+
+class LoggerBufferContainer
+{
+  public:
+    LoggerBufferContainer()
+        : currentIndex{0}, buffer{LoggerBuffer(circularBufferSize), LoggerBuffer(circularBufferSize)},
+          currentBuffer{buffer}, flushBuffer{buffer}
+    {}
+
+    LoggerBuffer *getFlushBuffer()
+    {
+        return flushBuffer;
+    }
+
+    LoggerBuffer *getCurrentBuffer()
+    {
+        return currentBuffer;
+    }
+
+    constexpr size_t getCircularBufferSize()
+    {
+        return circularBufferSize;
+    }
+
+    void nextBuffer()
+    {
+        ++currentIndex;
+        flushBuffer = currentBuffer;
+        currentIndex %= numberOfBuffers;
+        currentBuffer = &buffer[currentIndex];
+    }
+
+    size_t getCurrentIndex()
+    {
+        return currentIndex;
+    }
+
+  private:
+    static constexpr size_t circularBufferSize = 1000;
```

Comment:
1024 would look much better :D

Reply:
I wanted to leave the old buffer size, but honestly, it is not important. I'll change it to 1024 :)

Reply:
Done